src/address.rs

```diff
@@ -18,10 +17,54 @@ use core::ops::*;
 /// Unlike the naked underlying types, you can infallibly convert between,
 /// for example, an `Address<usize, ()>` and an `Address<u64, ()>` wherever
 /// such a conversion is lossless given the target CPU architecture.
-#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Copy, Clone, Default, PartialEq, Eq, PartialOrd, Ord)]
 #[repr(transparent)]
 pub struct Address<T, U>(T, PhantomData<U>);
 
+impl<T: core::fmt::Binary, U> core::fmt::Binary for Address<T, U> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        core::fmt::Binary::fmt(&self.0, f)
+    }
+}
+
+impl<T: core::fmt::LowerHex, U> core::fmt::Debug for Address<T, U> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.write_fmt(format_args!(
+            "Address(0x{:01$x})",
+            self.0,
+            size_of::<T>() * 2
```

Comment:
Why is this different from UpperHex?

Reply:
I think you read that wrong. That isn't `LowerHex`. That is `Debug`.

Reply:
oops.. right :)